src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,31 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.supplyAsync(() -> cache.get(userId), executor);
+        }
+        try {
```

Comment:
It's important to avoid using try-catch blocks that catch generic Exceptions like 'InterruptedException' without re-interrupting the thread. This can potentially swallow interrupt signals that are meant to stop the thread. Always handle 'InterruptedException' by calling 'Thread.currentThread().interrupt()' within the catch block or consider propagating the exception if it cannot be handled here.